5-appinfra/modules/env_baseline/db.tf

```diff
@@ -12,44 +12,88 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# CloudSQL Postgres instance
-module "cloudsql" {
-  source  = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
-  version = "~> 20.0"
-
-  for_each = toset(var.cluster_regions)
-
-  project_id = var.app_project_id
-  region     = each.value
-
-  name                = "${var.db_name}-${each.value}-${var.env}"
-  database_version    = "POSTGRES_14"
-  enable_default_db   = false
-  tier                = "db-custom-1-3840"
-  deletion_protection = false
-  availability_type   = "REGIONAL"
-
-  ip_configuration = {
-    ipv4_enabled                  = false
-    psc_enabled                   = true
-    psc_allowed_consumer_projects = [var.cluster_project_id]
+locals {
+  region = var.cluster_regions[0]
+}
+
+# Create alloydb cluster and instance.
+
+data "google_project" "project" {
+  project_id = var.network_project_id
+}
+
+module "alloydb" {
+  source  = "GoogleCloudPlatform/alloy-db/google"
+  version = "~> 3.0"
+
+  cluster_id       = "cluster-${local.region}-psc-${var.env}"
+  cluster_location = local.region
+  project_id       = var.app_project_id
+  cluster_initial_user = {
+    user     = "admin",
+    password = "admin"
   }
 
-  additional_databases = [
+  psc_enabled                   = true
+  psc_allowed_consumer_projects = [data.google_project.project.number]
+
+  primary_instance = {
+    instance_id        = "cluster-${local.region}-instance1-psc-${var.env}",
+    require_connectors = false
+    ssl_mode           = "ALLOW_UNENCRYPTED_AND_ENCRYPTED"
+  }
+
+  read_pool_instance = [
     {
-      name      = var.db_name
-      charset   = ""
-      collation = ""
-    },
+      instance_id        = "cluster-${local.region}-r1-psc-${var.env}"
+      display_name       = "cluster-${local.region}-r1-psc-${var.env}"
+      require_connectors = false
+      ssl_mode           = "ALLOW_UNENCRYPTED_AND_ENCRYPTED"
+    }
   ]
-  user_name     = "admin"
-  user_password = "admin" # this is a security risk - do not do this for real world use-cases!
 }
 
-resource "google_project_iam_member" "bank_of_anthos" {
-  for_each = toset(["roles/cloudsql.client", "roles/cloudsql.instanceUser"])
-  project  = var.app_project_id
-  role     = each.value
-  member   = "serviceAccount:bank-of-anthos@${var.cluster_project_id}.iam.gserviceaccount.com"
+resource "google_compute_network" "psc_vpc" {
+  name    = "psc-endpoint-vpc-${var.env}"
+  project = var.network_project_id
+}
+
+resource "google_compute_subnetwork" "psc_subnet" {
+  project       = var.network_project_id
+  name          = "psc-endpoint-subnet-${var.env}"
+  ip_cidr_range = "10.2.0.0/16"
+  region        = local.region
+  network       = google_compute_network.psc_vpc.id
+}
+
+# Create psc endpoing using alloydb psc attachment.
+
+resource "google_compute_address" "psc_consumer_address" {
```

Comment:
Should we create the IP Address in advance in 2-multitenant? https://github.com/GoogleCloudPlatform/terraform-google-enterprise-application/blob/main/2-multitenant/terraform.example.tfvars#L52

CC: @gtsorbo

Reply:
Yes, we should. removed it from this PR, leave its creation to the 2-multitenant